pilot/pkg/networking/plugin/plugin.go

```diff
@@ -54,6 +56,8 @@ func ModelProtocolToListenerProtocol(protocol config.Protocol) ListenerProtocol
 	case config.ProtocolTCP, config.ProtocolHTTPS, config.ProtocolTLS,
 		config.ProtocolMongo, config.ProtocolRedis, config.ProtocolMySQL:
 		return ListenerProtocolTCP
+	case config.ProtocolAuto:
+		return ListenerProtocolAuto
```

Comment:
I think we should get rid of the `config.ProtocolAuto` for now and instead have logic in the default case where you return `ListenerProtocolAuto` if the version is > 1.3.

This would also allow us to get rid of all of the version checks throughout the code, since no port will ever be configured to "auto" unless Envoy supports it.

@rshriram is that what you were thinking?

Reply:
Removed user-facing ProtocolAuto